src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -577,10 +568,9 @@ public static bool CanSupportCollectionInitializer(this ITypeSymbol typeSymbol,
             // SPEC: argument is more specific and no type argument is less specific than the
             // SPEC: corresponding type argument in the other.
 
-            var n1 = t1 as INamedTypeSymbol;
             var n2 = t2 as INamedTypeSymbol;
 
-            if (n1 == null)
+            if (t1 is not INamedTypeSymbol n1)
```

Comment:
fwiw: i think the old code was clearer here as it's operating on two values, and so the handling of each value is consistent.

Reply:
This was just random suggestion from VS. Ok, I'll revert this

Reply:
not all suggestions should be taken :)  they're jsut advising that you can write this, not that it necessarily should be written that way.  for things we have hard rules on, you get warnings/errors.
thanks!